coda_v2_python_client/firebase_client_wrapper.py

```diff
@@ -367,3 +367,86 @@ def set_segment_messages_metrics(self, segment_id, messages_metrics):
         :type metrics_map: core_data_modules.data_models.metrics.MessagesMetrics
         """
         self.get_segment_messages_metrics_ref(segment_id).set(messages_metrics.to_firebase_map())
+
+    def compute_segment_coding_progress(self, segment_id, messages=None, force_recount=False):
+        """
+        Compute and return the progress metrics for a given dataset.
+
+        This method will initialise the counts in Firestore if they do not already exist.
+
+        :param segment_id: Id of a segment.
+        :type segment_id: str
+        :param messages: list of core_data_modules.data_models.message.Message, defaults to None
+        :type messages: core_data_modules.data_models.message.Message | None
+        :param force_recount: If True, forces the recount of progress metrics. Defaults to False
+        :type force_recount: bool, optional
+        :return: Messages metrics.
+        :rtype: core_data_modules.data_models.metrics.MessagesMetrics
+        """
+        if not force_recount:
+            segment_messages_metrics = self.get_segment_messages_metrics(segment_id)
+            if segment_messages_metrics is not None:
+                return segment_messages_metrics
+
+        print(f"Performing a full recount of the metrics for segment {segment_id}...")
+        if messages is None:
+            messages = self.get_segment_messages(segment_id)
+        messages_with_labels = 0
+        wrong_scheme_messages = 0
+        not_coded_messages = 0
+
+        code_schemes = {code_scheme.scheme_id: code_scheme for code_scheme in self.get_all_code_schemes(segment_id)}
+
+        for message in messages:
+            # Get the latest label from each scheme
+            latest_labels = dict()  # of scheme id -> label
```

Comment:
I added a get_latest_labels function to core that you could use here

Reply:
updated